src/useField.ts

```diff
@@ -1,23 +1,23 @@
 import { useEffect } from 'react';
 import { useFormixContext } from './formixContext';
 
-// TODO: improve return type of the return and formix getField
-
 /**
  * A hook that helps you build custom fields in your form.
  *
  * It provides the properties of the field and a way to update its value.
  * @param name The name of the array field.
  */
-export default function useField(name: string) {
+export default function useField<T, Element extends HTMLInputElement>(
```

Comment:
make optional for the user to specify the element type

```suggestion
export default function useField<T, Element extends HTMLInputElement = HTMLInputElement>(
```